### PR TITLE
Migrated test execution syscall handler to runtime extensions

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
commit-id:4bf1ed90

---

**Stack**:
- #1228
- #1226
- #1219 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]
Really well abstracted! If we could get rid of that clone it would be perfect

**Comment by Arcticae** [comment]
@MaksymilianDemitraszek Can those 2 PRs be squashed? I think it'd give better visibility in case of those 2 PRs 
I will be reviewing this tmrw 🤓

File: crates/forge-runner/src/running.rs [hunk 1]
```diff
@@ -236,18 +237,22 @@ pub fn run_test_case(
...
+
     let contract_execution_syscall_handler =
         ContractExecutionSyscallHandler::wrap(cheatable_syscall_handler);
 
-    let mut test_execution_state = TestExecutionState {
-        environment_variables: &runner_params.environment_variables,
-        contracts: &runner_params.contracts,
+    let test_execution_state = TestExecutionState {
+        environment_variables: runner_params.environment_variables.clone(),
+        contracts: runner_params.contracts.clone(),
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/running.rs:246]: Good call, I forgot to add TODO here, will try to fix it

**Comment by MaksymilianDemitraszek** [comment]
@Arcticae Those are closely linked, and are touching different matter (one builds abstract types, and one uses them as an iterface), I don't really see a point squashing them. If you have a comments regarding use of generics and traits I would suggest commenting under the previous PR, If you have comments regarding the interface itself i would suggest addit it here. You have all PRs linked in the description for a reason

File: crates/forge-runner/src/forge_runtime_extension.rs [hunk 2]
```diff
@@ -716,48 +555,6 @@ struct ScarbStarknetContractArtifact {
     casm: Option<PathBuf>,
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/forge_runtime_extension.rs:555]: Please open an issue and don't make a mess under unrelated PR 🙏

File: crates/forge-runner/src/running.rs [hunk 3]
```diff
@@ -236,18 +237,22 @@ pub fn run_test_case(
...
     };
-    let mut test_execution_syscall_handler = TestExecutionSyscallHandler::wrap(
-        contract_execution_syscall_handler,
-        &mut test_execution_state,
-        &string_to_hint,
-    );
+
+    let mut forge_runtime = ExtendedRuntime(RuntimeExtension::<
```
Comment (Arcticae) [inline crates/forge-runner/src/running.rs:249]: Won't the type be inferred?

**Review by Arcticae — APPROVED** [review]

File: crates/forge-runner/src/running.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/running.rs]: Done

File: crates/forge-runner/src/forge_runtime_extension.rs [hunk 4]
```diff
@@ -13,296 +12,135 @@ use blockifier::execution::syscalls::{
...
-    fn run_resources(&self) -> &RunResources {
-        self.child
-            .child
-            .child
-            .context
-            .vm_run_resources
-            .run_resources()
-    }
-}
+// This runtime extenxion provides an implementation logic for functions from snforge_std library.
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/forge_runtime_extension.rs]: Done

File: crates/forge-runner/src/runtime.rs [hunk 5]
```diff
@@ -173,7 +181,7 @@ pub enum SyscallHandlingResult {
 #[derive(Debug)]
 pub enum CheatcodeHandlingResult {
     Forward,
-    Result(()), // TODO now use buffer later rewrite to return vector
+    Result(Vec<Felt252>),
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/runtime.rs:184]: hmm good call `Intercepted` ?

[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/runtime.rs:184]: I will also rename Forward to Forwarded

File: crates/forge-runner/src/running.rs (hunk 3 above)
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/running.rs:249]: It will be

Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/running.rs:249]: Done

File: crates/forge-runner/src/runtime.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/runtime.rs:184]: Done

File: crates/forge-runner/src/forge_runtime_extension.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (piotmag769) [reply crates/forge-runner/src/forge_runtime_extension.rs:556]: #1227

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**